dist/srpm/rhc.spec.in

```diff
@@ -0,0 +1,157 @@
+%define source_date_epoch_from_changelog 0
+
+# must be before %%gometa
+Version: @VERSION@
+
+# https://github.com/redhatinsights/rhc
+%global goipath     github.com/redhatinsights/rhc
+%global commit      @COMMIT@
+%global shortcommit %(c=%{commit}; echo ${c:0:7})
+%global date        %(date "+%Y%m%d")
+%global archivename rhc-%{version}
+
+%if %{has_go_rpm_macros}
+%gometa %{?fedora:-f}
+%else
+%global gourl https://github.com/RedHatInsights/%{name}
+%global gomodulesmode GO111MODULES=off
+%global gosource %{gourl}/releases/download/%{version}/rhc-%{version}.tar.gz
+%global gocompilerflags "-buildmode pie -compiler gc"
+%global scm git
+%forgemeta
+%endif
+
+# Manually redefine %%dist to work around an issue in COPR where the build root
+# that creates the srpm does not define a value for %%dist. This should *NOT* be
+# carried in downstream; this is strictly an upstream/COPR/CI workaround.
+%if "%{dist}" == ""
+%global dist %{distprefix}.fc%{fedora}
+%endif
+
+%if 0%{?fedora}
+%global setup_flags -Dvendor=False
+%else
+%endif
+%global setup_flags -Dvendor=True
+
+%global common_description %{expand:
+rhc is a client that registers a system with RHSM and activates the Red Hat yggd
+MQTT client.}
+
+%global golicenses      LICENSE
+%global godocs          CONTRIBUTING.md README.md
+
+Name:    rhc
+Release: 99%{?dist}
+Epoch:   1
+Summary: Client for registering Red Hat Enterprise Linux systems
+
+License: GPL-3.0-only
+URL:     %{gourl}
+Source0: %{gosource}
+
+Requires:      insights-client
+Requires:      yggdrasil >= 0.4
+Requires:      yggdrasil-worker-package-manager
+Requires:      subscription-manager
+
+BuildRequires:  systemd-rpm-macros
+BuildRequires:  meson
+BuildRequires:  pkgconfig(systemd)
+BuildRequires:  pkgconfig(bash-completion)
+BuildRequires:  golang >= 1.18
+
+
+%description %{common_description}
+
+%package compat
+Summary: Transition package to support migrating from rhcd to yggd
+
+Requires:   yggdrasil >= 0.4.2
+Recommends: rhc
+
+%description compat
+Transition package to support migrating from rhcd to yggd.
+
+%if %{has_go_rpm_macros}
+%gopkg
+%endif
+
+%prep
+%if 0%{?fedora}
+%goprep
+%else
+%autosetup
+%endif
+
+%if 0%{?fedora}
+%generate_buildrequires
+%go_generate_buildrequires
+%endif
+
+%build
+%undefine _auto_set_build_flags
+export %gomodulesmode
+%{?gobuilddir:export GOPATH="%{gobuilddir}:${GOPATH:+${GOPATH}:}%{?gopath}"}
+ls
+pwd
+%meson %setup_flags "-Dgobuildflags=[%(echo %{expand:%gocompilerflags} | sed -e s/"^"/"'"/ -e s/" "/"', '"/g -e s/"$"/"'"/), '-tags', '"rpm_crashtraceback\ ${BUILDTAGS:-}"', '-a', '-v', '-x']" -Dgoldflags='%{?currentgoldflags} -B 0x%(head -c20 /dev/urandom|od -An -tx1|tr -d " \n") -compressdwarf=false -linkmode=external -extldflags "%{build_ldflags} %{?__golang_extldflags}"'
+%meson_build
+
+%install
+%meson_install
+install --directory %{buildroot}%{_unitdir}
+install --directory %{buildroot}%{_sysconfdir}/rhc
+ln -sf yggdrasil.service        %{buildroot}%{_unitdir}/rhcd.service
```

Comment:
The file `rhcd.service` is also part of `rhc` RPM package (not `rhc-compat` RPM package). Is it desired behavior? If not, then I think that we want to have `%install compat` section.

Similar comment is about `config.toml` file.

Reply:
Oh interesting. I didn't notice that. I'll add a `%install compat` section and move the symlink there. I only see `config.toml` in the file list in `rhc-compat`. I don't see it in `rhc`.

#### rhc
```
/usr/bin/rhc
/usr/lib/.build-id
/usr/lib/.build-id/3d
/usr/lib/.build-id/3d/e47d8103ecf1c1356b7f7a932e2c8b8afff9de
/usr/lib/systemd/system-preset/80-rhc.preset
/usr/lib/systemd/system/rhc-canonical-facts.service
/usr/lib/systemd/system/rhc-canonical-facts.timer
/usr/lib/systemd/system/rhcd.service
/usr/share/bash-completion/completions/rhc
/usr/share/doc/rhc
/usr/share/doc/rhc/CONTRIBUTING.md
/usr/share/doc/rhc/README.md
/usr/share/licenses/rhc
/usr/share/licenses/rhc/LICENSE
/usr/share/man/man1/rhc.1.gz
```

#### rhc-compat
```
/etc/rhc/config.toml
/usr/lib/systemd/system/rhcd.service
```

Reply:
Oh, it's because in `%files`, the line `%{_unitdir}/*` exists.

Reply:
This has been corrected.

Reply:
> Oh, it's because in `%files`, the line `%{_unitdir}/*` exists.

Yeah, it marked `rhcd.service` to be part of `rhc` RPM too.